app/ui/src/main/java/com/fsck/k9/fragment/AccountRetriever.kt

```diff
@@ -0,0 +1,16 @@
+package com.fsck.k9.fragment
+
+import android.database.Cursor
+import com.fsck.k9.Account
+import com.fsck.k9.Preferences
+import com.fsck.k9.fragment.MLFProjectionInfo.ACCOUNT_UUID_COLUMN
+
+class AccountRetriever constructor(
+        private val preferences: Preferences
+) : (Cursor) -> Account {
+
+    override fun invoke(cursor: Cursor): Account {
```

Comment:
In my opinion the operator overloading doesn't make the code easier to read. On the contrary, it's harder to figure out what's going on. Especially for people coming from Java.

Also, it feels like this is a helper function at best. The implementation is highly coupled to the used projection. So it probably shouldn't live in its own class.

Reply:
Removed, agreed. Data unpacking logic will get moved out of Adapter/ViewHolder at some point. 

Class was a poor attempt at removing the fragment dependency, which was evident once I started again at ViewHolder refactor.